### PR TITLE
refactor: use single Oracle class for Price Oracles

### PR DESCRIPTION
**Pull request by khancode**
## High Level Overview of Change

Use single Oracle class for Price Oracles support.

### Context of Change

- Use single `Oracle` class for `get_aggregate_price` and `ledger_entry` to resolve duplicate name conflict.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### Did you update CHANGELOG.md?

- [ ] Yes
- [x] No, this change does not impact library users

## Test Plan

Updated unit tests.

### CODE REVIEW

File: xrpl/models/requests/get_aggregate_price.py [hunk 1]
```diff
@@ -53,7 +51,7 @@ def _get_errors(self: GetAggregatePrice) -> Dict[str, str]:
 
 @require_kwargs_on_init
 @dataclass(frozen=True)
-class Oracle(TypedDict):
+class OracleEntry:
```
Comment (mvadari) [inline xrpl/models/requests/get_aggregate_price.py:54]: Would this be confusing given that the other `Oracle` class is in `LedgerEntry`?

Naming is hard 😕

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Construction of Wallet throws an "Invalid Seed" error, if the secret is not decode-able.
 - Rectify the incorrect usage of a transaction flag name: Update `TF_NO_DIRECT_RIPPLE` to `TF_NO_RIPPLE_DIRECT`
 - Add the missing `AMMDeposit` Flag `TF_TWO_ASSET_IF_EMPTY`
+- Minor refactor changes for Price Oracles support.
```
Comment (mvadari) [inline CHANGELOG.md:21]: The changelog doesn't need to be updated here if it's in the same version that the price oracle PR is in.

File: xrpl/models/requests/get_aggregate_price.py (hunk 1 above)
Comment (khancode) [reply xrpl/models/requests/get_aggregate_price.py:54]: Yeah not sure 😅 We could rename it to `OracleParamEntry` or `OracleParam`?

Comment (mvadari) [reply xrpl/models/requests/get_aggregate_price.py:54]: Looks like they're identical, so why not just have one object?

Comment (khancode) [reply xrpl/models/requests/get_aggregate_price.py:54]: Not quite identical in types: one has `oracle_document_id` set to `Union[str, int]` and the other has `int`. But I guess using `Union[str, int]` would be sufficient.

Comment (mvadari) [reply xrpl/models/requests/get_aggregate_price.py:54]: I'm guessing `Union[str, int]` is probably correct for the RPC too, tbh.

Comment (mvadari) [reply xrpl/models/requests/get_aggregate_price.py:54]: https://github.com/XRPLF/rippled/issues/5014

Comment (khancode) [reply xrpl/models/requests/get_aggregate_price.py]: Done in https://github.com/XRPLF/xrpl-py/pull/704/commits/56f1260af99e5101de24eb6c31d6c60f4caa4a57

**Review by mvadari — APPROVED** [review]

**Review by pdp2121 — APPROVED** [review]

**Pull request merged**